Adds integration test framework for e2e tests

bharathkkb opened the pull request:
- Adds test setup
- Adds fixtures for following stages
    - bootstrap
    - org
    - env

Todo:

- [x]  CB yaml

- [x] CB Trigger https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/734

rjerrems commented:
Overall structure is looking good from my point of view @bharathkkb

bharathkkb commented on 1-org/org_policy.tf:
```diff
@@ -149,9 +149,13 @@ module "org_enforce_bucket_level_access" {
-  parent_id   = local.organization_id
-  policy_name = "default policy"
+  parent_id   = var.org_id
+  policy_name = "default policy-${random_id.suffix.hex}"
```
@rjerrems I am having some test failures [1] for this `access_context_manager_policy` resource. Looks like in one of the previous runs [2] a destroy step failed leaving this behind. I initially thought  I could bypass with a rand suffix, but it doesnt seem dependent on the policy. Any ideas on how we can prevent this in the future or clean up? 

[1]: https://pantheon.corp.google.com/cloud-build/builds/9cfa6a60-7055-4826-b5f7-57fd5fd26b0e;step=5?project=cloud-foundation-cicd
[2]: https://pantheon.corp.google.com/cloud-build/builds/fd9ecc79-f5aa-4c72-bc82-1f6ce6fb095f?project=cloud-foundation-cicd

rjerrems replied:
Yeah I think we need to make sure this resource is cleaned up, even if it requires some gcloud commands. An org can only have one access context manager policy from my testing with gcloud commans

bharathkkb replied:
gotcha, sounds good.

rjerrems commented on 0-bootstrap/main.tf:
```diff
@@ -43,15 +43,18 @@ resource "google_folder" "seed" {
-  source                  = "terraform-google-modules/bootstrap/google"
-  version                 = "~> 1.0"
+  source = "github.com/terraform-google-modules/terraform-google-bootstrap"
+  # version                 = "~> 1.0"
```
is this supposed to be commented out?

bharathkkb replied:
@rjerrems this was temporary as I was waiting for the new 1.2 bootstrap release, will revert.

rjerrems replied:
It looks like this is still commented out?

bharathkkb replied:
just fixed :)

bharathkkb commented:
@rjerrems this should be ready for review, I will add some kitchen tests in a subsequent PR.

rjerrems commented on 1-org/org_policy.tf:
```diff
@@ -149,9 +149,13 @@ module "org_enforce_bucket_level_access" {
-  parent_id   = local.organization_id
-  policy_name = "default policy"
+  parent_id   = var.org_id
+  policy_name = "default policy-${random_id.suffix.hex}"
```
nit: the random suffice is not really required for this resource given that only one can exist in an org

bharathkkb replied:
removed

rjerrems commented on build/int.cloudbuild.yaml:
```diff
@@ -1,4 +1,4 @@
-# Copyright 2020 Google LLC
```
nit: this should be 2020

rjerrems replied:
There are a few instances of this FYI

bharathkkb replied:
done

rjerrems commented on test/fixtures/org/main.tf:
```diff
@@ -0,0 +1,28 @@
+  org_id                    = var.org_id
+  billing_account           = var.billing_account
+  terraform_service_account = var.terraform_sa_email
+  default_region            = "us-east4"
```
nit: it would be good to have some consistency with the regions used

bharathkkb replied:
using `us-central1` for consistency with other fixtures

bharathkkb commented:
@rjerrems I have addressed the comments but tests are failing due to a [Github packages outage](https://www.githubstatus.com/).  I will re run them when its back.

rjerrems commented:
I just triggered the builds again, looks like there may be some errors based on the region for BigQuery - I don't think you can specify us-central1

bharathkkb commented:
@rjerrems issue was here https://github.com/terraform-google-modules/terraform-example-foundation/pull/65#discussion_r453456499 thats probably why i went with `"us-east4"`. Reverting to that now.
